subsys/net/ip/net_shell.c

```diff
@@ -3532,6 +3532,140 @@ static int cmd_net_ipv6(const struct shell *shell, size_t argc, char *argv[])
+			snprintk(remaining_str, sizeof(remaining_str) - 1,
+				 "infinite");
+		} else {
+			snprintk(remaining_str, sizeof(remaining_str) - 1,
+				 "%u", (uint32_t)(remaining / 1000U));
+		}
+
+		PR("%s  \t%s\t%s    \t%12s/%12s\n",
+		       addrtype2str(ipv4->unicast[i].addr_type),
+		       addrstate2str(ipv4->unicast[i].addr_state),
+		       remaining_str,
+		       net_sprint_ipv4_addr(
+			       &ipv4->unicast[i].address.in_addr),
+		       net_sprint_ipv4_addr(
+			       &ipv4->netmask));
+	}
+}
+#endif /* CONFIG_NET_NATIVE_IPV4 */
+
+static int cmd_net_ipv4(const struct shell *shell, size_t argc, char *argv[])
+{
+	struct net_shell_user_data user_data;
+
+	PR("IPv4 support                              : %s\n",
+	   IS_ENABLED(CONFIG_NET_IPV4) ?
+	   "enabled" : "disabled");
+	if (!IS_ENABLED(CONFIG_NET_IPV4)) {
+		return -ENOEXEC;
+	}
+
+#if defined(CONFIG_NET_NATIVE_IPV4)
+	PR("IPv4 fragmentation support                : %s\n",
+	   IS_ENABLED(CONFIG_NET_IPV4_FRAGMENT) ? "enabled" :
+	   "disabled");
+	PR("Max number of IPv4 network interfaces "
+	   "in the system          : %d\n",
+	   CONFIG_NET_IF_MAX_IPV4_COUNT);
+	PR("Max number of unicast IPv4 addresses "
+	   "per network interface   : %d\n",
+	   CONFIG_NET_IF_UNICAST_IPV4_ADDR_COUNT);
+	PR("Max number of multicast IPv4 addresses "
+	   "per network interface : %d\n",
+	   CONFIG_NET_IF_MCAST_IPV4_ADDR_COUNT);
+
+	user_data.shell = shell;
+	user_data.user_data = NULL;
+
+	/* Print information about address lifetime */
+	net_if_foreach(ip_address_lifetime_cb, &user_data);
+#endif
+
+	return 0;
+}
+
+static int cmd_net_ip_add(const struct shell *shell, size_t argc, char *argv[])
+{
+#if defined(CONFIG_NET_NATIVE_IPV4)
+	struct net_if *iface = NULL;
+	int idx;
+	struct in_addr addr;
+
+	if (argv[1]) {
+		idx = get_iface_idx(shell, argv[1]);
+		if (idx < 0) {
+			return -ENOEXEC;
+		}
+
+		iface = net_if_get_by_index(idx);
+		if (!iface) {
+			PR_WARNING("No such interface in index %d\n", idx);
+			return -ENOEXEC;
+		}
+	if (argv[2]) {
+		if (net_addr_pton(AF_INET, argv[2], &addr)) {
+			PR_ERROR("Invalid address: %s\n", argv[2]);
+			return -EINVAL;
+			}
+#if defined(CONFIG_NET_DHCPV4)
+		/* In case DHCP is enabled, make the static address tentative,
+		 * to allow DHCP address to override it. This covers a usecase
+		 * of "there should be a static IP address for DHCP-less setups",
+		 * but DHCP should override it (to use it, NET_IF_MAX_IPV4_ADDR
+		 * should be set to 1). There is another usecase: "there should
+		 * always be static IP address, and optionally, DHCP address".
+		 * For that to work, NET_IF_MAX_IPV4_ADDR should be 2 (or more).
+		 * (In this case, an app will need to bind to the needed addr
+		 * explicitly.)
+		 */
+		net_if_ipv4_addr_add(iface, &addr, NET_ADDR_OVERRIDABLE, 0);
+#else
+		net_if_ipv4_addr_add(iface, &addr, NET_ADDR_MANUAL, 0);
+#endif
+		return 0;
+		}
+		return -1;
+	}
+#endif /* CONFIG_NET_NATIVE_IPV4 */
```

Comment:
In case native IPv4 is disabled,  I think the command should print an error to shell.

Reply:
Done